### PR TITLE
LP-2831 - Integrate segment for ADG

### PR DESCRIPTION
**Pull request by aightmunam**
[LP-2831](https://philanthropyu.atlassian.net/browse/LP-2831)

[Theme PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/475)

***

#### Description:

- Integrate Segment for ADG
- Segment track and identify for all registration, application, webinar, courseware and admin events

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2786/workflows/workflows_id/jobs/2786/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
common/djangoapps/student/models.py (75.0%): Missing lines 1361
lms/djangoapps/grades/models.py (66.7%): Missing lines 661-662
openedx/adg/admin_dashboard/applications/views.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/analytics.py (80.3%): Missing lines 56,63-65,72-75,182-183,185-186
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/handlers.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_bu_prereq_courses_status.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_program_prereq_courses_status.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/conftest.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
openedx/adg/lms/applications/tests/test_analytics.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/applications/tests/test_views.py (100%)
openedx/adg/lms/applications/views.py (100%)
openedx/adg/lms/registration_extension/analytics.py (0.0%): Missing lines 4,6,9,16,22-23
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/registration_extension/tests/conftest.py (100%)
openedx/adg/lms/registration_extension/tests/test_models.py (100%)
openedx/adg/lms/webinars/analytics.py (100%)
openedx/adg/lms/webinars/api/views.py (100%)
openedx/adg/lms/webinars/constants.py (100%)
openedx/adg/lms/webinars/tests/test_analytics.py (100%)
openedx/core/djangoapps/user_authn/views/register.py (50.0%): Missing lines 85,249
-------------
Total:   366 lines
Missing: 23 lines
Coverage: 93%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2808/workflows/workflows_id/jobs/2808/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
common/djangoapps/student/models.py (75.0%): Missing lines 1361
lms/djangoapps/grades/models.py (66.7%): Missing lines 661-662
openedx/adg/admin_dashboard/applications/views.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/analytics.py (80.3%): Missing lines 56,63-65,72-75,182-183,185-186
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/handlers.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_bu_prereq_courses_status.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_program_prereq_courses_status.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/conftest.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
openedx/adg/lms/applications/tests/test_analytics.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/applications/tests/test_views.py (100%)
openedx/adg/lms/applications/views.py (100%)
openedx/adg/lms/registration_extension/analytics.py (0.0%): Missing lines 4,6,9,16,22-23
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/registration_extension/tests/conftest.py (100%)
openedx/adg/lms/registration_extension/tests/test_models.py (100%)
openedx/adg/lms/webinars/analytics.py (100%)
openedx/adg/lms/webinars/api/views.py (100%)
openedx/adg/lms/webinars/constants.py (100%)
openedx/adg/lms/webinars/tests/test_analytics.py (100%)
openedx/core/djangoapps/user_authn/views/register.py (50.0%): Missing lines 85,249
-------------
Total:   366 lines
Missing: 23 lines
Coverage: 93%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - Edx
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2816/workflows/workflows_id/jobs/2816/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/master...HEAD, staged and unstaged changes
-------------
common/djangoapps/student/models.py (75.0%): Missing lines 1361
lms/djangoapps/grades/models.py (66.7%): Missing lines 661-662
openedx/adg/admin_dashboard/applications/views.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/analytics.py (80.3%): Missing lines 56,63-65,72-75,182-183,185-186
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/handlers.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_bu_prereq_courses_status.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_program_prereq_courses_status.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/conftest.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
openedx/adg/lms/applications/tests/test_analytics.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
openedx/adg/lms/applications/tests/test_views.py (100%)
openedx/adg/lms/applications/views.py (100%)
openedx/adg/lms/registration_extension/analytics.py (0.0%): Missing lines 4,6,9,16,22-23
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/registration_extension/tests/conftest.py (100%)
openedx/adg/lms/registration_extension/tests/test_models.py (100%)
openedx/adg/lms/webinars/analytics.py (100%)
openedx/adg/lms/webinars/api/views.py (100%)
openedx/adg/lms/webinars/constants.py (100%)
openedx/adg/lms/webinars/tests/test_analytics.py (100%)
openedx/core/djangoapps/user_authn/views/register.py (50.0%): Missing lines 85,249
-------------
Total:   366 lines
Missing: 23 lines
Coverage: 93%
-------------
```

**Review by annas-usman — APPROVED** [review]

File: openedx/adg/lms/applications/models.py [hunk 1]
```diff
@@ -154,6 +154,29 @@ def program_courses_status_text(self):
...
+    def application_progress(self):
+        """
+        Return the current progress of the application i.e Not Started, Started,
+        Submitted (No decision yet), Waitlisted and Accepted.
+
+        Returns:
+            str: Returns the current application progress for a user
+        """
+        application = getattr(self.user, 'application', None)
+        application_status = getattr(application, 'status', None)
```
Comment (asadali145) [inline openedx/adg/lms/applications/models.py:167]: Why using getattr?

File: common/djangoapps/student/models.py [hunk 2]
```diff
@@ -1355,6 +1357,9 @@ def update_enrollment(self, mode=None, is_active=None, skip_refund=False):
         if activation_changed:
             if self.is_active:
                 self.emit_event(EVENT_NAME_ENROLLMENT_ACTIVATED)
+                if not is_testing_environment():
+                    emit_course_progress_event(self.user, CourseOverview.get_from_id(self.course_id))
+
```
Comment (asadali145) [inline common/djangoapps/student/models.py:1362]: Actually we recently changes our way to handle this kind of change, We write a helper in our adg code and simply call that here and move this test env check in that helper this way only single line will change in core file. For now we can leave it this way. Just mentioning for future reference.

File: openedx/core/djangoapps/user_authn/views/register.py [hunk 3]
```diff
@@ -242,7 +243,10 @@ def create_account_with_params(request, params):
         except Exception:  # pylint: disable=broad-except
             log.exception(u"Enable discussion notifications failed for user {id}.".format(id=user.id))
 
-    _track_user_registration(user, profile, params, third_party_provider)
+    if is_testing_environment():
+        _track_user_registration(user, profile, params, third_party_provider)
+    else:
+        emit_registration_event(user)
```
Comment (asadali145) [inline openedx/core/djangoapps/user_authn/views/register.py:249]: Same here.

**Review by asadali145 — COMMENTED** [review]
Great Work! Just a few things, It is merged but I had a few points, Can we please note and maybe in some improvements ticket, We can improve these?

**Pull request merged**